libraries/common/include/common/default_construct_copyable_movable.hpp

```diff
@@ -14,13 +14,14 @@ struct DefaultConstructCopyableMovable {
   T val;
 
   DefaultConstructCopyableMovable() = default;
-  DefaultConstructCopyableMovable(T val) : val(std::move(val)) {}
+  ~DefaultConstructCopyableMovable() = default;
```

Comment:
why does it need destructor?

Reply:
the rule is that as soon as you declare any of the 5 ctors/operators, you should declare all of them